### PR TITLE
Add enroll command

### PR DESCRIPTION
**Pull request by FlyingDonkeys**
- enroll command basically adds an SCA into the storage, after a person and course has been added
- if there is duplicate tutorial, SCA is not added

### CODE REVIEW

**Comment by FlyingDonkeys** [comment]
- Issues: For some reason creating multiple SCAs causes the previously created SCA to disappear. maybe @miloaisdino can take a look?

**Comment by Zeepheru** [comment]
Is this going to be v1.5 or v1.4?

---

Also, with regards to the current **Merge Conflicts**,

### CliSyntax.java
```java
    public static final Prefix PREFIX_TAG = new Prefix("t/");
<<<<<<< add_sca_commands
    public static final Prefix PREFIX_COURSE_CODE = new Prefix("c/");
    // t/ prefix is used by tags...
    public static final Prefix PREFIX_TUTORIAL_ID = new Prefix("tut/");
=======
    public static final Prefix PREFIX_CODE = new Prefix("c/");
    public static final Prefix PREFIX_TUTORIAL = new Prefix("tut/");
>>>>>>> master
```
- Just keep both sets of fields, we'll come in and remove the duplicate later; otherwise solving it here seems like it would have to change a lot
  - [ ] Set up as issue todo
  
### Tutorial.java
- Either is fine, both have the goal of fixing a spacing issue in the original message anyway

### Messages.java::format(StudentCourseAssiciation)
- Unless I'm mistaken, using `SCA::toString()` directly will dump out the full Person `toString()`
- The idea with my version of `format(SCA)` was to shrink the output to just Matric Number, Course, and Tut Id - you decide which is more ideal ig

File: src/main/java/tahub/contacts/logic/commands/EnrollCommand.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            throw new CommandException(Messages.MESSAGE_PERSON_NOT_FOUND);
+        }
+
+        Course dummyCourse = new Course(courseCodeToAdd, "dummy course name");
+        if (!model.hasCourse(dummyCourse)) {
+            throw new CommandException(Messages.MESSAGE_COURSE_NOT_FOUND);
+        }
+
+        // Since addressbook contains a student with this matric number, this operation is guaranteed to succeed
+        Person personInSca = model.getAddressBook().getPersonList().stream()
```
[the remark this answers is not in the document]
Comment (FlyingDonkeys) [reply src/main/java/tahub/contacts/logic/commands/EnrollCommand.java:92]: oh rly? maybe you can show me the method later after class? thanks

File: src/main/java/tahub/contacts/model/studentcourseassociation/StudentCourseAssociationList.java [hunk 2]
```diff
@@ -209,7 +205,7 @@ public String toString() {
     /**
      * Returns true if {@code sca} contains only unique sca.
      */
-    private boolean scaAre(List<StudentCourseAssociation> sca) {
+    private boolean scaAreUnique(List<StudentCourseAssociation> sca) {
```
Comment (accountexeregister) [inline src/main/java/tahub/contacts/model/studentcourseassociation/StudentCourseAssociationList.java:208]: Should case `sca` to a better name, like `scas` or `scaList`

File: src/main/java/tahub/contacts/model/course/Course.java [hunk 3]
```diff
@@ -71,12 +71,12 @@ public boolean equals(Object other) {
...
 
-        return courseCode.equals(otherCourse.courseCode) && courseName.equals(otherCourse.courseName);
+        return courseCode.equals(otherCourse.courseCode);
     }
 
     @Override
     public int hashCode() {
-        return courseName.hashCode();
+        return courseCode.hashCode();
     }
```
Comment (accountexeregister) [inline src/main/java/tahub/contacts/model/course/Course.java:80]: I understand you are changing `equals` to `hashCode` to implement checking if `Course`  with `courseCode` already exists in the SCA list. However, I think equality and hashcode should take into account both `courseCode` and `courseName`. Maybe you can try seeing if the check method exists in `Model` or you can implement it yourself by looping through the SCA list and seeing if a course with `courseCode` equal to a given `courseCode` already exists

**Comment by FlyingDonkeys** [comment]
Also when i update any field in Person/Course, i.e: edit name of a person. Any related sca with that person inside must reflect the new name but that doesnt seem to be happening rn. @miloaisdino any idea how to handle that?

**Review by Zeepheru — APPROVED** [review]
LGTM!!

**Comment by FlyingDonkeys** [comment]
LGTM

**Review by accountexeregister — APPROVED** [review]
LOOKS AMAZING

**Pull request merged**